client/packages/sprotty-client/src/features/layout/layout-commands.ts

```diff
@@ -0,0 +1,416 @@
+         */
+        public readonly dimension: ResizeDimension = ResizeDimension.Width,
+        /**
+         * Function to reduce the dimension to a target dimension value, see Reduce.* for potential functions.
+         */
+        public readonly reductionFunction: (...values: number[]) => number) {
+    }
+}
+
+export enum Alignment {
+    Left,
+    Center,
+    Right,
+    Top,
+    Middle,
+    Bottom
+}
+
+export namespace Select {
+    export function all(elements: SelectableBoundsAware[]) {
+        return elements;
+    }
+
+    export function first(elements: SelectableBoundsAware[]) {
+        return [elements[0]];
+    }
+
+    export function last(elements: SelectableBoundsAware[]) {
+        return [elements[elements.length - 1]];
+    }
+}
+
+
+export class AlignElementsAction implements Action {
+    readonly kind = AlignElementsCommand.KIND;
+
+    constructor(
+        /**
+         * IDs of the elements that should be aligned. If no IDs are given, the selected elements will be aligned.
+         */
+        public readonly elementIds: string[] = [],
+        /**
+         * Alignment direction.
+         */
+        public readonly alignment: Alignment = Alignment.Left,
+        /**
+         * Function to selected elements that are considered during alignment calculation, see Select.* for potential functions.
+         */
+        public readonly selectionFunction: (elements: SelectableBoundsAware[]) => SelectableBoundsAware[] = Select.all) {
+    }
+}
+
+abstract class LayoutElementsCommand extends Command {
+    constructor(@inject(TYPES.Action) protected action: ResizeElementsAction | AlignElementsAction,
+        @inject(TYPES.IActionDispatcher) protected actionDispatcher: IActionDispatcher,
+        @inject(GLSP_TYPES.SelectionService) protected selectionService: SelectionService) {
+        super();
+    }
+
+    getActionElements(context: CommandExecutionContext): SelectableBoundsAware[] {
+        const model = context.root;
+        const elementIDs = this.action.elementIds;
+        if (elementIDs.length === 0) {
+            // collect the selected elements from the selection service (selection order is kept by service)
+            this.selectionService.getSelectedElementIDs().forEach(elementID => elementIDs.push(elementID));
+        }
+        const selectableBoundsAware: SelectableBoundsAware[] = [];
+        elementIDs.forEach(id => {
+            const element = model.index.getById(id);
+            if (element && isNonRoutableSelectedBoundsAware(element)) {
+                selectableBoundsAware.push(element);
+            }
+        });
+        return selectableBoundsAware;
+    }
+
+    dispatchAction(action: Action) {
+        this.actionDispatcher.dispatch(action);
+    }
+
+    dispatchActions(actions: Action[]) {
+        this.actionDispatcher.dispatchAll(actions);
+    }
+}
+
+export class ResizeElementsCommand extends LayoutElementsCommand {
+    static readonly KIND = 'layout:resize';
```

Comment:
According to sprotty guidelines, commands should be annotated with `@injectable`

Reply:
Done